File: src/main/scala/com/barley/respository/FileRepository.scala
package com.barley.respository

import java.lang.Long

import com.barley.model.File
import org.springframework.data.repository.CrudRepository
import org.springframework.stereotype.Component

/**
  * Created by vikram.gulia on 11/25/15.
  */
@Component
trait FileRepository extends CrudRepository[File, Long]

File: src/main/java/com/barley/repository/FileRepository.java
package com.barley.repository;

import com.barley.model.Attachment;
import com.barley.model.File;
import org.springframework.data.repository.CrudRepository;
import org.springframework.stereotype.Component;

import java.util.List;

@Component
public interface FileRepository extends CrudRepository<File, Long> {
    List<File> findByLongFileName(String longFileName);
}
